addons/website/static/tests/tours/media_iframe_video.js

```diff
@@ -59,3 +59,49 @@ wTourUtils.registerWebsitePreviewTour(
         },
     ]
 );
+
+wTourUtils.registerWebsitePreviewTour(
+    "website_snippet_background_video",
+    {
+        test: true,
+        url: "/",
+        edition: true,
+    },
+    [
+        wTourUtils.dragNDrop({
+            id: "s_text_block",
+            name: "Text",
+        }),
+        {
+            content: "Click on the text block.",
+            trigger: "iframe #wrap section.s_text_block",
+        },
+        {
+            content: "Click on the 'Background Video' button option.",
+            trigger: "we-button[data-name='bg_video_toggler_opt']",
+        },
+        {
+            content: "Click on the first sample video in the modal.",
+            trigger: "#video-suggestion .o_sample_video",
+        },
+        {
+            content: "Check the video is select.",
+            trigger: "textarea.is-valid",
+            run: () => {}, // This is a check.
+        },
+        {
+            content: "Click on the 'Add' button to apply the selected video as the background.",
+            trigger: ".modal-footer button:contains('Add')",
+        },
+        {
+            content: "Verify that the video is set as the background of the snippet.",
+            trigger: "iframe #wrap section.s_text_block",
+            run: function () {
+                const isBackgroundVideo = this.$anchor[0].classList.contains("o_background_video");
+                if (!isBackgroundVideo) {
+                    console.error("Background video is not set on the snippet.");
+                }
+            },
+        },
```

Comment:
Instead of this `run`, can't you just add the o_background_video class in the trigger?

Reply:
@qsm-odoo  Yes sir, we can directly use o_background_video in the class. However, if someone fails this test, it will display the failed steps instead of what should be there in case of failure. It might take time to understand why the test was written this way.

For that person, it will be helpful to know if the failure occurred because the background video was not found. Both approaches work the same, so it's up to you. I'll proceed as you suggest.

Please let me know your point of view.

Thanks you for your review!

Reply:
I understand your point of view but in the end, the tour system was designed to not have to write `run` function like that everytime we want to check something is there. I would go with the least amount of code. I think it will be as easy for the person seeing "Tour step `Verify that the video is set as the background of the snippet.` failed" to understand that the video may not have been added properly, even though, indeed, the step could fail for another reason :wink: 

There is actually another reason: the way you did it, I would not be surprised if a race condition could occur. You last step is true **immediately** (there is the block in the page already, you are editing it), meaning that technically, your `run` function could be called while the video is not applied **yet**. That's probably not the case as we (I hope) apply the video before closing the modal but if the code evolves... this might break for no reason.
=> What you should really be saying is "wait for the video to be there", thanks to an unique trigger, and the test fails if it never comes.